Tests

Tests are failing: python3.10, Fedora OS...

self = <niaarm.dataset.Dataset object at 0x7fff8c25bac0>
    def __analyse_types(self):
        r"""Extract data types for the data in a dataset."""
        for head in self.header:
            col = self.data[head]
    
            if col.dtype == "float":
                dtype = "float"
                min_value = col.min()
                max_value = col.max()
                unique_categories = None
            elif col.dtype == "int":
                dtype = "int"
                min_value = col.min()
                max_value = col.max()
                unique_categories = None
            else:
                dtype = "cat"
>               unique_categories = sorted(col.astype('string').unique().tolist(), key=str.lower)
E               AttributeError: 'StringArray' object has no attribute 'tolist'
niaarm/dataset.py:44: AttributeError

--- a/niaarm/dataset.py
+++ b/niaarm/dataset.py
@@ -41,7 +41,7 @@ class Dataset:
                 unique_categories = None
             else:
                 dtype = "cat"
-                unique_categories = sorted(col.astype('string').unique().tolist(), key=str.lower)
+                unique_categories = sorted(col.astype(str).unique().tolist(), key=str.lower)
                 min_value = None
                 max_value = None
 

--- a/niaarm/tests/test_coverage.py
+++ b/niaarm/tests/test_coverage.py
@@ -3,6 +3,7 @@ from niaarm.association_rule import AssociationRule
 from niaarm.dataset import Dataset
 import os
 
+
 class TestCoverage(TestCase):
     # let's borrow a test case from wikipedia:
     # https://en.wikipedia.org/wiki/Lift_(data_mining)

--- a/niaarm/tests/test_cut_point.py
+++ b/niaarm/tests/test_cut_point.py
@@ -3,6 +3,7 @@ from niaarm.association_rule import AssociationRule, _cut_point
 from niaarm.dataset import Dataset
 import os
 
+
 class TestCutPoint(TestCase):
     # let's borrow a test case from Wikipedia:
     # https://en.wikipedia.org/wiki/Lift_(data_mining)

--- a/niaarm/tests/test_read_csv.py
+++ b/niaarm/tests/test_read_csv.py
@@ -2,6 +2,7 @@ from unittest import TestCase
 from niaarm.dataset import Dataset
 import os
 
+
 class TestReadCSVAbalone(TestCase):
     def test_read_features(self):
 

--- a/niaarm/tests/test_rule_building.py
+++ b/niaarm/tests/test_rule_building.py
@@ -3,6 +3,7 @@ from niaarm.association_rule import AssociationRule, _rule_feasible, _get_permut
 from niaarm.dataset import Dataset
 import os
 
+
 class TestBuildRuleA(TestCase):
     # let's borrow a test case from Wikipedia:
     # https://en.wikipedia.org/wiki/Lift_(data_mining)

--- a/niaarm/tests/test_shrinkage.py
+++ b/niaarm/tests/test_shrinkage.py
@@ -3,6 +3,7 @@ from niaarm.association_rule import AssociationRule, _normalize, _cut_point
 from niaarm.dataset import Dataset
 import os
 
+
 class TestShrinkageA(TestCase):
     def setUp(self):
         data = Dataset(os.path.join(os.path.dirname(__file__), 'test_data', 'wiki_test_case.csv'))

--- a/niaarm/tests/test_support_confidence.py
+++ b/niaarm/tests/test_support_confidence.py
@@ -3,6 +3,7 @@ from niaarm.association_rule import AssociationRule, _cut_point
 from niaarm.dataset import Dataset
 import os
 
+
 class TestSupportConfidence(TestCase):
     # let's borrow test case from wikipedia:
     # https://en.wikipedia.org/wiki/Lift_(data_mining)

